PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -25,60 +25,101 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 #include <Library/ResetSystemLib.h>
 #include <Library/SecureBootLib.h>
 #include <Library/BootloaderCommonLib.h>
-#include <Library/LiteFvLib.h>
 #include <Library/FirmwareUpdateLib.h>
 #include <Guid/SystemResourceTable.h>
 #include "FirmwareUpdateHelper.h"
 
 /**
   Verify the firmware version to make sure it is no less than current firmware version.
 
-  @param[in]  Stage1ABase   Pointer to stage 1A base.
-  @param[in]  IsFd          Does Stage1ABase point to Stage1A FD
-                            or SBL Stage1A FV ?
-  @param[out] Version       Pointer to version of the firmware
+  @param[in] ImageHdr     Pointer to the fw mgmt capsule image header
+  @param[in] FwPolicy     Firmware update policy.
 
-  @retval  EFI_SUCCESS        The operation completed successfully.
-  @retval  others             There is error happening.
+  @retval  EFI_SUCCESS    The operation completed successfully.
+  @retval  others         There is error happening.
 **/
 EFI_STATUS
-GetVersionfromFv (
-  IN  UINT32              *Stage1ABase,
-  IN  BOOLEAN             IsFd,
-  OUT BOOT_LOADER_VERSION **Version
+VerifySblVersion (
+  IN  EFI_FW_MGMT_CAP_IMAGE_HEADER  *ImageHdr,
+  IN  FIRMWARE_UPDATE_POLICY  FwPolicy
   )
 {
-  EFI_STATUS                  Status;
-  EFI_FFS_FILE_HEADER         *FfsFile;
-  EFI_FIRMWARE_VOLUME_HEADER  *FvHeader;
+  UINT32                CompBase;
+  UINT32                CompSize;
+  BOOT_LOADER_VERSION   *CapsuleBlVersion;
+  BOOT_LOADER_VERSION   *CurrentBlVersion;
+  EFI_STATUS            Status;
 
-  FvHeader = (EFI_FIRMWARE_VOLUME_HEADER *)(UINTN)(*Stage1ABase);
   //
-  // Stage 1A FD has FSPT FV first, so move on to the next FV
+  // SVN check for BIOS
   //
-  if (IsFd) {
-    FvHeader = (EFI_FIRMWARE_VOLUME_HEADER *)((UINTN)FvHeader + (UINTN)FvHeader->FvLength);
-  }
+  CapsuleBlVersion = NULL;
+  CurrentBlVersion = NULL;
 
   //
-  // Get version info FFS from FV
+  // Get svn from existing system firmware
   //
-  Status = GetFfsFileByName(FvHeader,&gBootLoaderVersionFileGuid,&FfsFile);
+  Status = GetSvn (0xFFFFFFFC, &CurrentBlVersion);

```

Comment:
0xFFFFFFFC is not component base.   The contents at this location is Stage1A base.

Reply:
Updated variable name accordingly.